ui/src/components/swap/IncomingOffers.tsx

```diff
@@ -0,0 +1,32 @@
+import { usePurse } from '../../hooks/usePurse';
+import { useContractStore } from '../../store/contract';
+import { stringifyData } from '../../utils/stringify';
+import IncomingOffer from './IncomingOffer';
+
+const IncomingOffers = () => {
+  const { instances } = useContractStore();
+  const swaparooInstance = instances?.['swaparoo'];
+  const invitationPurse = usePurse('Invitation');
```

Comment:
if ```swaparooInstance``` instance, and ```invitationPurse``` these are parameterized, i can def see this as a reusable component. Maybe we already have something like this in ui-kit? Would love your thoughts. The UI is nice & elegant too 🚀